File: Python/Punto_16.py
"""Punto 16: Escriba una función que cuente el número de números positivos de una lista.

Ejemplo: cuentaPositivos([0,1,-3,-2,8,-1,6]) = 3"""
def cuentaPositivos(lista):
  return len(list(filter(lambda elemento: elemento>0,lista)))
lista=[0,1,-3,-2,8,-1,6]
print(cuentaPositivos(lista))

File: Java/Punto16.java
package Java;

import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;
import java.util.stream.Stream;

// No lo he aclarado pero ingreso Lista, como estoy trabajando en el mismo paquete, la interfaz
// ingreso Lista solo se ha declarado en el punto 13 con filter, y la vamos a reutilizar hasta el final.


@FunctionalInterface
interface ContadorPositivos {
    long contar(List<Integer> lista);
}

public class Punto16 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        IngresoLista ingresoLista = () -> {
            System.out.println("Ingrese los números separados por espacios:");
            return Stream.of(scanner.nextLine().split(" "))
                    .map(Integer::parseInt)
                    .collect(Collectors.toList());
        };
        List<Integer> entrada = ingresoLista.obtenerLista();
        scanner.close();
        ContadorPositivos contadorPositivos = lista -> lista.stream()
                .filter(n -> n > 0)
                .count();

        System.out.println("Cantidad de números positivos: " + contadorPositivos.contar(entrada));
    }
}